src/core/server/saved_objects/migrationsv2/model.ts

```diff
@@ -495,11 +499,37 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
           outdatedDocuments: res.right.outdatedDocuments,
           lastHitSortValue: res.right.lastHitSortValue,
         };
+      } else {
+        // we don't have any more outdated documents but had issues with transforming docs, we fail the migration
+        if (stateP.corruptDocumentIds.length > 0 || stateP.transformErrors.length > 0) {
+          // if documents couldn't be transformed or there were transformation errors we fail the migration
+          const corruptDocumentIdReason =
+            stateP.corruptDocumentIds.length > 0
+              ? `The following saved object documents are corrupt: ${stateP.corruptDocumentIds.join(
+                  ','
+                )}`
+              : '';
+          const transformErrorsReason =
+            stateP.transformErrors.length > 0
+              ? 'The following saved object documents could not be transformed:/n' +
+                stateP.transformErrors
+                  .map((errObj) => `${errObj.rawId}: ${errObj.err.message}`)
+                  .join('/n')
```

Comment:
I've extracted the error message construction from out of the main model body and in so doing realized the implementations were slightly different 😱 The error message format should be the same now, since we call the same method with similar arguments.